### PR TITLE
Update base64 to 0.21

### PR DESCRIPTION
**Pull request by Ayush1325**
Also added once_cell for caching the new Engine based APIs of base64

Signed-off-by: Ayush Singh <ayushsingh1325@gmail.com>

### CODE REVIEW

**Comment by Ayush1325** [comment]
Possibly a duplicate of #123 although the versions are different.

**Review by seanmonstar — APPROVED** [review]
Thanks!

**Comment by seanmonstar** [comment]
Ah, I guess base64 decided to use the 2021 edition, requiring a newer compiler. Sigh. What version is needed?

**Comment by Ayush1325** [comment]
> Ah, I guess base64 decided to use the 2021 edition, requiring a newer compiler. Sigh. What version is needed?

I think it's [1.57.0](https://github.com/marshallpierce/rust-base64/blob/master/RELEASE-NOTES.md#breaking-changes-2). Should I update this PR or would you like to do that separately?

**Comment by nicoxxl** [comment]
Beware, hyper MSRV is 1.56 : https://github.com/hyperium/hyper/blob/master/docs/MSRV.md

But 1.57.0 was released a little more than one year ago : https://blog.rust-lang.org/2021/12/02/Rust-1.57.0.html

**Comment by serprex** [comment]
@Ayush1325 please go ahead & update minrust in `.github/workflows/ci.yml`

**Comment by serprex** [comment]
@seanmonstar tungstenite opted to switch from base64 crate to data-encoding given how extensive latest changes are: https://github.com/snapview/tungstenite-rs/commit/c6769e3daebc064f1e1e711f552c7970a42d8346

would you prefer a PR replacing base64 with data-encoding?

**Comment by serprex** [comment]
Looks like `base64` went ahead & bumped MSRV to 1.60 with 0.21.1: https://github.com/marshallpierce/rust-base64/pull/238

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: CI
 on: [push, pull_request]
 
 env:
-  minrust: 1.51.0
+  minrust: 1.57.0
```
Comment (serprex) [inline .github/workflows/ci.yml:6]:
```suggestion
  minrust: 1.51.0
```
This can be reverted now that base64 0.21.3 has MSRV 1.47

**Comment by serprex** [comment]
@seanmonstar with MSRV issues seeming to be resolved for base64, can this PR progress?

**Comment by seanmonstar** [comment]
Yea absolutely, let's do it :)

**Comment by seanmonstar** [comment]
Huh, well looks like with `httpdate`, we need an MSRV of 1.56 anyways. 🤷

File: Cargo.toml [hunk 2]
```diff
@@ -19,7 +19,8 @@ members = [
 [dependencies]
 http = "0.2.0"
 headers-core = { version = "0.2", path = "./headers-core" }
-base64 = "0.13"
+base64 = "0.21.3"
+bitflags = "1.0"
```
Comment (serprex) [inline Cargo.toml:23]:
```suggestion
```

**Pull request merged**